templates/_helpers.tpl

```diff
@@ -457,9 +457,12 @@ Sets the injector deployment update strategy
 {{/*
 Sets extra pod annotations
 */}}
-{{- define "vault.annotations" -}}
-  {{- if .Values.server.annotations }}
+{{- define "vault.annotations" }}
       annotations:
+  {{- if .Values.server.configAnnotation }}
+        vaultproject.io/config-checksum: {{ include "vault.config" . | sha256sum }}
```

Comment:
For a recent k8s auth plugin [feature](https://github.com/hashicorp/vault-plugin-auth-kubernetes/commit/daeaf8df3497b60b1d0804b5c0d396c462e46b2e), we went with `vault.hashicorp.com/` as the annotation prefix, and we should probably match that here. I believe @thyton checked a few other examples for consistency before landing on that, cc'ed to keep me honest :)

```suggestion
        vault.hashicorp.com/config-checksum: {{ include "vault.config" . | sha256sum }}
```